ide/models/files.py

```diff
@@ -239,28 +240,55 @@ class SourceFile(TextFile):
     )
     target = models.CharField(max_length=10, choices=TARGETS, default='app')
 
+    DIR_MAP = {
+        # Using an OrderedDict here ensures that 'src/' is checked last in get_details_for_path().
+        'native': OrderedDict([
+            ('pkjs', os.path.join('src', 'js')),
+            ('worker', 'worker_src'),
+            ('app', 'src')
+        ]),
+        'pebblejs': {
+            'app': os.path.join('src', 'js')
+        },
+        'simplyjs': {
+            'app': 'src'
+        },
+        'package': {
+            'app': os.path.join('src', 'c'),
+            'public': 'include',
+            'pkjs': os.path.join('src', 'js')
+        }
+    }
+
+    @classmethod
+    def get_details_for_path(cls, project_type, path):
```

Comment:
JSYK for your review: I backported some new changes from the rocky branch to this branch.